### PR TITLE
Handle unapplied remote move - initialScan - sync

### PR DESCRIPTION
**Pull request by aenario**
This PR includes and fix the test from #1235

### CODE REVIEW

File: core/local/chokidar_event.js [hunk 1]
```diff
@@ -2,12 +2,13 @@
...
 import type fs from 'fs'
+import type {Metadata} from '../metadata'
 
 export type ChokidarAdd = {type: 'add', path: string, stats: fs.Stats}
 export type ChokidarAddDir = {type: 'addDir', path: string, stats: fs.Stats}
 export type ChokidarChange = {type: 'change', path: string, stats: fs.Stats}
-export type ChokidarUnlink = {type: 'unlink', path: string}
-export type ChokidarUnlinkDir = {type: 'unlinkDir', path: string}
+export type ChokidarUnlink = {type: 'unlink', path: string, old?: Metadata}
+export type ChokidarUnlinkDir = {type: 'unlinkDir', path: string, old?: Metadata}
```
Comment (sebn) [inline core/local/chokidar_event.js]: Isn't this supposed to be what chokidar gives us? How come it haves *old* Metadata?

Comment (sebn) [reply core/local/chokidar_event.js]: (ok answer in code below)

**Comment by aenario** [comment]
Cleanup : rebased & removed unused function `pretendAddFromMetadata`

**Comment by sebn** [comment]
I would have expected this to be handled at the Merge level, no in the watcher: It makes the watcher more complex, with more *made up* events (the real move events were already merged) and it will have to be reimplemented for the new AtomWatcher... Also I wonder whether it could happen on the remote side...

**Comment by sebn** [comment]
As discussed with @nono, applying pending changes on restart before initial scan would be a bad idea (since we have no guarantee the local FS still matches the meta tree), handling it in the Merge would be far more complicated than the current solution (according to the current way the Merge works), so better keep it in the local watcher for now (even if it means reimplementing it for the new atom watcher).

So +1 to merge after release 3.10.3

**Comment by aenario** [comment]
[Si un move est appliqué dans le pouch mais pas sur le FS (application interrompue), le redémmarrage de l'app crée un cas de conflits infinis](https://trello.com/c/g5ltJ01O/1191-si-un-move-est-appliqu%C3%A9-dans-le-pouch-mais-pas-sur-le-fs-application-interrompue-le-red%C3%A9mmarrage-de-lapp-cr%C3%A9e-un-cas-de-conflits)

**Pull request merged**